test/lib/clients/vault_test.rb

```diff
@@ -1,43 +1,51 @@
 require "test_helper"
 
 class VaultTest < ActiveSupport::TestCase
-  attr_reader :random_mount
+  attr_reader :intermediate_ca_mount
+  attr_reader :root_ca_mount
 
   setup do
     @client = Clients::Vault
-    @random_mount = SecureRandom.hex(4)
+    @root_ca_mount = SecureRandom.hex(4)
+    @intermediate_ca_mount = SecureRandom.hex(4)
   end
 
   teardown do
-    vault_client.sys.unmount(random_mount)
+    vault_client.sys.unmount(root_ca_mount)
+    vault_client.sys.unmount(intermediate_ca_mount)
   end
 
   test "#configure_kv" do
-    @client.stub :kv_mount, random_mount do
+    @client.stub :kv_mount, intermediate_ca_mount do
       assert @client.configure_kv
       engines = vault_client.sys.mounts
-      assert_equal "kv", engines[random_mount.to_sym].type
+      assert_equal "kv", engines[intermediate_ca_mount.to_sym].type
     end
   end
 
   test "#configure_pki" do
-    @client.stub :intermediate_ca_mount, random_mount do
-      assert @client.configure_pki
-      engines = vault_client.sys.mounts
-      assert_equal "pki", engines[random_mount.to_sym].type
-
-      read_cert = vault_client.logical.read("#{random_mount}/cert/ca").data[:certificate]
-      assert_match "BEGIN CERTIFICATE", read_cert
-
-      cluster_config = vault_client.logical.read("#{random_mount}/config/cluster").data
-      assert_equal "#{vault_addr}/v1/#{random_mount}", cluster_config[:path]
-      assert_equal "#{vault_addr}/v1/#{random_mount}", cluster_config[:aia_path]
-
-      role_config = vault_client.logical.read("#{random_mount}/roles/astral").data
-      assert_not_nil role_config[:issuer_ref]
-      assert_equal 720.hours, role_config[:max_ttl]
-      assert_equal true, role_config[:allow_any_name]
-     end
+    @client.stub :root_ca_mount, root_ca_mount do
+      @client.stub :intermediate_ca_mount, intermediate_ca_mount do
+        assert @client.configure_pki
+
+        [ root_ca_mount, intermediate_ca_mount ].each do |mount|
+          engines = vault_client.sys.mounts
+          assert_equal "pki", engines[mount.to_sym].type
+
+          read_cert = vault_client.logical.read("#{mount}/cert/ca").data[:certificate]
+          assert_match "BEGIN CERTIFICATE", read_cert
+
+          cluster_config = vault_client.logical.read("#{mount}/config/cluster").data
+          assert_equal "#{vault_addr}/v1/#{mount}", cluster_config[:path]
+          assert_equal "#{vault_addr}/v1/#{mount}", cluster_config[:aia_path]
+        end
+
+        role_config = vault_client.logical.read("#{intermediate_ca_mount}/roles/astral").data
+        assert_not_nil role_config[:issuer_ref]
+        assert_equal 720.hours, role_config[:max_ttl]
+        assert_equal true, role_config[:allow_any_name]
+      end
```

Comment:
should we also test for the root role?

Reply:
removed the root role -- I don't think we need